packages/playwright-core/src/server/socksClientCertificatesInterceptor.ts

```diff
@@ -274,6 +276,13 @@ export class ClientCertificatesProxy {
     loadDummyServerCertsIfNeeded();
   }
 
+  getProxyAgent(host: string, port: number) {
+    const proxyFromEnv = getProxyForUrl(`https://${host}:${port}`);
+    if (proxyFromEnv)
+      return createProxyAgent({ server: proxyFromEnv });
+    return createProxyAgent(this._proxy);
```

Comment:
If the user config specifies a proxy explicitly, we should prioritize that, similar to how the browser would use it instead of the env var.